doc/specs/#888 - Com Api.md

```diff
@@ -0,0 +1,889 @@
+    runtimeclass GetCompositeAppCatalogOptions
+    {
+        GetCompositeAppCatalogOptions(); 
+
+        /// Get a composite catalog to allow searching a user defined or pre defined source
+        /// and a local source (Installing, Installed) together  
+        /// IMPLEMENTATION NOTE: Windows Package Manager currently only supports 
+        /// one local source and one remote source per composite catalog.
+        /// DESIGN NOTE: It seems likely that in the future we will want to support 
+        /// letting callers use two local sources (i.e. both Installing and Installed)
+        IVector<AppCatalog> Catalogs { get; };
+        /// The local source(s)
+        LocalAppCatalog LocalAppCatalog;
+        /// Sets the default search behavior if the catalog is a composite catalog.
+        CompositeSearchBehavior CompositeSearchBehavior;
+    }
+
+    [contract(Microsoft.Management.Deployment.WindowsPackageManagerContract, 1)]
+    enum AppInstallScope
+    {
+        User,
+        Machine,
+    };
+
+    [contract(Microsoft.Management.Deployment.WindowsPackageManagerContract, 1)]
+    enum AppInstallMode
+    {
+        /// The default experience for the installer. Installer may show some UI.
+        Default,
+        /// Runs the installer in silent mode. This suppresses the installer's UI to the extent 
+        /// possible (installer may still show some required UI).
+        Silent,
+        /// Runs the installer in interactive mode.
+        Interactive,
+    };
+
+    /// Options when installing a package.
+    /// Intended to allow full compatibility with the "winget install" command line interface.
+    [contract(Microsoft.Management.Deployment.WindowsPackageManagerContract, 1)]
+    runtimeclass InstallOptions
+    {
+        InstallOptions();
+
+        /// DESIGN NOTE: --Exact from the winget install command line is implied here by virtue of the find 
+        /// then install object semantics.
+        /// Likewise --Query from the command line is not implemented here since the query is required up 
+        /// front in order to get the CatalogPackage object.
+        /// DESIGN NOTE: Using PackageVersionInfo rather than CatalogPackage felt like it would be too 
+        /// specific here. CatalogPackage and PackageVersionId (or if unspecified then the latest one) may be nicer
+        /// if we end up having versions that have different applicability or other more complex logic about 
+        /// which version is preferred.
+        CatalogPackage CatalogPackage;
+        /// Optionally specifies the version from the package to install. If unspecified the version matching 
+        /// CatalogPackage.GetLatestVersion() is used.
+        PackageVersionId PackageVersionId;
+
+        /// Specifies alternate location to install package (if supported).
+        String PreferredInstallLocation;
+        /// User or Machine. 
+        AppInstallScope AppInstallScope;
+        /// Silent, Interactive, or Default

```

Comment:
I assume Default is the default.

Reply:
Default here is more like "installer defined", so we could choose a different default, but that would be fine too.

Reply:
DISCUSSION: If the app requires interactive it'll pass it in.  Many apps will want Silent.  This is about how much UI the installer itself will show.